staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -266,6 +294,12 @@ func (kr *kubeRegistry) enableHiddenStableCollectors() {
 	kr.CustomMustRegister(cs...)
 }
 
+func (kr *kubeRegistry) Reset() {
```

Comment:
Why not take the lock here? At least a read lock?

Reply:
ack